File: 프로그래머스/unrated/132267. 콜라 문제/콜라 문제.js
function solution(a, b, n) {
    let getCoke = 0;
    let answer = 0;
    let remain = 0
    while(true){
        for( n ; n > 1 ; n--){
            if(n % a === 0){ 
                getCoke = (n / a)*b; // case1:10/5/2/1/1  case2:6/2/1
                answer += (n / a)*b;
                n = remain + getCoke; // case1:10/5/3/2/1   case2:8/4/2
                if(n < a){
                    break;
                }
                n++;
                remain = 0 //
        		}else{
           		 remain++
   		 }
	}
    return answer;
	}
}

File: 프로그래머스/unrated/132267. 콜라 문제/콜라 문제.py
def solution(a, b, n):
    i=0
    j=0
    sum=0
    while(True):
        i=n//a*b # 교환한 콜라
        j=n%a # 남은 콜라
        n=i+j 
        sum+=i
        if n<a:
            break

    return sum